File: app/src/main/java/com/bigscreen/mangindo/network/service/MangaNetworkService.java
package com.bigscreen.mangindo.network.service;

import com.bigscreen.mangindo.network.model.response.ChapterListResponse;
import com.bigscreen.mangindo.network.model.response.MangaContentListResponse;
import com.bigscreen.mangindo.network.model.response.NewReleaseResponse;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface MangaNetworkService {

    @GET("main.php")
    Observable<NewReleaseResponse> getNewReleaseManga();

    @GET("chapter_list.php")
    Observable<ChapterListResponse> getChapters(@Query("manga") String mangaTitle);

    @GET("image_list.php")
    Observable<MangaContentListResponse> getMangaContents(@Query("manga") String mangaTitle, @Query("chapter") String chapter);

}

File: app/src/main/java/com/bigscreen/mangindo/network/service/MangaNetworkService.kt
package com.bigscreen.mangindo.network.service

import com.bigscreen.mangindo.network.model.response.ChapterListResponse
import com.bigscreen.mangindo.network.model.response.MangaContentListResponse
import com.bigscreen.mangindo.network.model.response.NewReleaseResponse
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface MangaNetworkService {

    @GET("main.php")
    fun getNewReleaseManga(): Observable<NewReleaseResponse>

    @GET("chapter_list.php")
    fun getChapters(
            @Query("manga") mangaTitle: String
    ): Observable<ChapterListResponse>

    @GET("image_list.php")
    fun getMangaContents(
            @Query("manga") mangaTitle: String,
            @Query("chapter") chapter: String
    ): Observable<MangaContentListResponse>
}
